salt/modules/file.py

```diff
@@ -1651,8 +1651,13 @@ def replace(path,
         with salt.utils.fopen(path,
                               mode='r',
                               buffering=bufsize) as r_file:
+            count_replaced = 0
             for line in r_file:
-                result, nrepl = re.subn(cpattern, repl, line, count)
+                if count == 0 or count_replaced != count:
```

Comment:
To clarify, this would be my alternative code:

```python
count_replaced = count
for line in r_file:
    if count == 0:
        # replace all occurrences
        result, nrepl = re.subn(cpattern, repl, line, count)
    elif count_replaced > 0:
        # replace up to `count` instances
        result, nrepl = re.subn(cpattern, repl, line, count_replaced)
        count_replaced -= nrepl
    else:
        # already replaced `count` instances
        break
```

The write block is basically the same but we don't want to break because we need to finish writing out the entire file. Maybe this:

```python
count_replaced = count
for line in r_file:
    if count == 0:
        # replace all occurrences
        result, nrepl = re.subn(cpattern, repl, line, count)
    elif count_replaced > 0:
        # replace up to `count` instances
        result, nrepl = re.subn(cpattern, repl, line, count_replaced)
        count_replaced -= nrepl
    else:
        # already replaced `count` instances
        result = line
    try:
        w_file.write(result)
    # ... snip ...
```